engine/common/synchronization/engine.go

```diff
@@ -271,10 +295,17 @@ func (e *Engine) processAvailableResponses() {
 			continue
 		}
 
-		msg, ok = e.pendingBlockResponses.Get()
+		msg, ok = e.pendingBatchResponses.Get()
 		if ok {
-			e.onBlockResponse(msg.OriginID, msg.Payload.(*messages.BlockResponse))
-			e.metrics.MessageHandled(metrics.EngineSynchronization, metrics.MessageBlockResponse)
+			e.onBatchResponse(msg.OriginID, msg.Payload.(*messages.BatchResponse))
+			e.metrics.MessageHandled(metrics.EngineSynchronization, metrics.MessageBatchResponse)
+			continue
+		}
+
+		msg, ok = e.pendingRangeResponses.Get()
```

Comment:
Shouldn't we prioritize range responses? Blocks in range requests typically need to be processed before batch requests. (Likewise for the collection node sync engine)

Reply:
Yes, that's correct. Thanks